tests/setuptab.test.js

```diff
@@ -580,4 +587,260 @@ describe('Form drag-and-drop', () => {
+    const setupForm = await findByTestId('setup-form');
+    const setupInput = await findByLabelText(RegExp(`${spec.args.arg1.name}`));
+
+    const fileDragEvent = createEvent.dragEnter(setupInput);
+    // `dataTransfer.files` normally returns a `FileList` object. Since we are
+    // defining our own dataTransfer.files we are also creating an object
+    // with properties that mimic FileList object
+    const fileValue = {};
+    Object.defineProperties(fileValue, {
+      path: { value: 'foo.txt' },
+      length: { value: 1 },
+    });
+    Object.defineProperty(fileDragEvent, 'dataTransfer', {
+      value: { files: [fileValue] }
+    });
+    fireEvent(setupInput, fileDragEvent);
+
+    expect(setupForm).not.toHaveClass("dragging");
+    expect(setupInput).toHaveClass("input-dragging");
+
+    const fileDropEvent = createEvent.drop(setupInput);
+    Object.defineProperty(fileDropEvent, 'dataTransfer', {
+      value: { files: [fileValue] }
+    });
+    fireEvent(setupInput, fileDropEvent);
+
+    expect(setupInput).not.toHaveClass("input-dragging");
+    expect(setupForm).not.toHaveClass("dragging");
+    expect(setupInput).toHaveValue( 'foo.txt' );
+  });
+
+  test('Drag enter/leave of a file sets .input-dragging class on input', async () => {
+    const spec = {
+      module: `natcap.invest.${MODULE}`,
+      args: {
+        arg1: {
+          name: 'Workspace',
+          type: 'directory',
+        },
+        arg2: {
+          name: 'AOI',
+          type: 'vector',
+        },
+      },
+    };
+    fetchValidation.mockResolvedValue(
+      [[Object.keys(spec.args), 'invalid because']]
+    );
+
+    const { findByLabelText, findByTestId } = renderSetupFromSpec(spec);
+    const setupInput = await findByLabelText(RegExp(`${spec.args.arg1.name}`));
+
+    const fileDragEnterEvent = createEvent.dragEnter(setupInput);
+    // `dataTransfer.files` normally returns a `FileList` object. Since we are
+    // defining our own dataTransfer.files we are also creating an object
+    // with properties that mimic FileList object
+    const fileValue = {};
+    Object.defineProperties(fileValue, {
+      path: { value: 'foo.txt' },
+      length: { value: 1 },
+    });
+    Object.defineProperty(fileDragEnterEvent, 'dataTransfer', {
+      value: { files: [fileValue] }
+    });
+    fireEvent(setupInput, fileDragEnterEvent);
+
+    expect(setupInput).toHaveClass("input-dragging");
+
+    const fileDragLeaveEvent = createEvent.dragLeave(setupInput);
+    Object.defineProperty(fileDragLeaveEvent, 'dataTransfer', {
+      value: { files: [fileValue] }
+    });
+    fireEvent(setupInput, fileDragLeaveEvent);
+
+    expect(setupInput).not.toHaveClass("input-dragging");
+  });
+
+  test('Drag and drop on a disabled input element.', async () => {
+    const spec = {
+      module: `natcap.invest.${MODULE}`,
+      args: {
+        arg1: {
+          name: 'Workspace',
+          type: 'directory',
+          disabled: true,

```

Comment:
Yep, after #84 you can pass a boolean `enabled` prop to `ArgInput`s. For a couple of tests I mocked out the UI spec file like this
```
beforeAll(() => {
    const mockUISpec = {FooModel: {order: ["foo", "bar"]}};
    jest.mock('../src/ui_config', () => mockUISpec);
}
afterAll(() => {
    jest.resetModules();
}
```

Reply:
What was I looking at? 🙃

Reply:
I ultimately went with using the current `uiSpec` setup, knowing that it will change... I did not go with rendering `ArgInput` because it relies on handlers from `ArgsForm`. https://github.com/natcap/invest-workbench/pull/92/commits/744114de22a529559e2d1273d1697074bbf05b30